provider/cmd/pulumi-gen-google-native/main.go

```diff
@@ -149,12 +150,30 @@ func copyJSONFile(url, outDir, path string) error {
 		return err
 	}
 
-	var ifce interface{}
+	ifce := map[string]interface{}{}
 	err = json.Unmarshal(body, &ifce)
 	if err != nil {
 		return err
 	}
 
+	if rev, ok := ifce["revision"].(string); ok {
+		f, err := os.Open(filepath.Join(outDir, path))
+		if err == nil {
+			existing := map[string]interface{}{}
+			if err := json.NewDecoder(f).Decode(&existing); err != nil {
+				return err
+			}
+			existingRev, ok := existing["revision"].(string)
+			if ok && existingRev > rev {
```

Comment:
Is this comparison valid? It looks like you're comparing two strings, so this would be the lexical ordering. I'm not sure if we'd need to parse the date?

Reply:
The revision is in `YYYYMMDD` format which allows for lexical ordering. You can see from some of the diffs in the PR that it does have the desired effect.